index.js

```diff
@@ -1,37 +1,72 @@
 import { createSelector } from 'reselect';
 
+// @TODO Move it into a separate file and test it
+function flatCacheObjectCreator() {
```

Comment:
I hardly understand why not use classes here. It's exactly what they are for. If you don't like the code generated by Babel, you can always use plain old prototypes, which is just the same and a much more natural approach. I'd personally stick to classes though.